### DISCUSSION TITLE
client-preset is generating wrong type definition for operations that use fragments

### DISCUSSION THREAD

**Discussion by giancarlosisasi** [dotansimha/graphql-code-generator#8639] [Q&A] [answered]
### Which packages are impacted by your issue?

- @graphql-codegen/client-preset

### Describe the bug

Hi there, I'm trying to migrate from apollo-client to graphql-code-generator and we found this "_bug_"

When using the following graphql operations:
````gql
#import "@my-org/web/dist/Graphql/Fragments/userFragment.graphql"
query MeAuthContextAuthQuery {
  me {
    ...UserFragment
  }
}
````
````gql
# content of @my-org/web/dist/Graphql/Fragments/userFragment.graphql"
fragment UserFragment on MeNode {
  id
  originalId
  email
  segmentation
  ...
}
````

After running the codegen, the `me` field type doesn't have all the Fragment fields.

The generated type is:
````ts
export type MeAuthContextAuthQuery = { __typename?: 'Query', me?: (
    { __typename?: 'MeNode' }
    & { ' $fragmentRefs'?: { 'UserFragmentFragment': UserFragmentFragment } }
  ) | null };
````

Which makes the following code fails:

````ts
const {data} = useQuery(MeAuthContextAuthDocument)
const email = data?.me?.email; // => email doesn't exists in me object :(
````

### Your Example Website or App

https://codesandbox.io/s/awesome-lucy-ck9kwo?file=/test.tsx

### Steps to Reproduce the Bug or Issue

I have added a minimal reproduction case using codesanbox.

1. Go to https://codesandbox.io/s/awesome-lucy-ck9kwo?file=/test.tsx
2. Open the test.tsx file
3. A typescript error is shown because email doesn't exists in user type 😢 

### Expected behavior

The type generated when using a fragment should contain all the fields defined in the fragment.

### Platform

- OS: macOS
- NodeJS: 14.18
- `graphql` version: 15.5
- `@graphql-codegen/*` version(s): 
````json
"@graphql-codegen/cli": "2.13.12",
"@graphql-codegen/client-preset": "1.1.4",
"@graphql-typed-document-node/core": "3.1.1",
````

### Codegen Config File

This is the real config that we are using in our project. For simplicity, the codesanbox contains a minimal config.

````ts
const config: CodegenConfig = {
  overwrite: true,
  ignoreNoDocuments: true,
  generates: {
    'packages/auth/src/gql-auth/': {
      schema: GRAPH_ENDPOINT_AUTH,
      documents: 'packages/auth/src/**/*.auth.graphql',
      preset: 'client',
      plugins: [],
    },
    'packages/auth/src/gql/': {
      schema: GRAPH_ENDPOINT,
      documents: [
        'packages/auth/src/**/*.graphql',
        'packages/auth/node_modules/@my-org/web/dist/**/*.graphql',
        '!packages/auth/node_modules/@my-org/web/dist/**/*.*.graphql',
        '!packages/auth/src/**/*.auth.graphql',
      ],
      preset: 'client',
      plugins: [],
    },
  },
};
````

**Comment by saihaj** [comment]
It’s because with client preset it is doing fragment masking. https://the-guild.dev/graphql/codegen/plugins/presets/gql-tag-operations-preset#fragment-masking

**Comment by giancarlosisasi** [comment]
@saihaj do u know how to disable that? I can't find the docs related to the client preset😢

**Comment by jantimon** [comment, marked as answer]
It's already visible in the source:

https://github.com/dotansimha/graphql-code-generator/blob/4e3cfdce1bc86c576d792fe634c537017129f682/packages/presets/client/src/index.ts#L24-L36

**Reply by giancarlosisasi** [reply]
thanks @jantimon !
